modules/core/src/core/lib/layer-manager.js

```diff
@@ -181,88 +173,75 @@ export default class LayerManager {
   }
 
   getViews() {
-    return this.views;
+    return this.viewManager.views;
   }
 
   // Get a set of viewports for a given width and height
-  // TODO - Intention is for deck.gl to autodeduce width and height and drop the need for props
   getViewports() {
-    this._rebuildViewportsFromViews();
-    this.context.viewport = this.viewports[0];
-    return this.viewports;
+    const viewports = this.viewManager.getViewports();
+    this.context.viewport = viewports[0];
+    return viewports;
   }
 
   /**
-   * Set parameters needed for layer rendering and picking.
+   * Set props needed for layer rendering and picking.
    * Parameters are to be passed as a single object, with the following values:
    * @param {Boolean} useDevicePixels
    */
-  /* eslint-disable complexity */
-  setParameters(parameters) {
-    if ('eventManager' in parameters) {
-      this._initEventHandling(parameters.eventManager);
+  /* eslint-disable complexity, max-statements */
+  setProps(props) {
+    if ('eventManager' in props) {
+      this._initEventHandling(props.eventManager);
     }
 
-    if (
-      'pickingRadius' in parameters ||
-      'onLayerClick' in parameters ||
-      'onLayerHover' in parameters
-    ) {
-      this._setEventHandlingParameters(parameters);
+    if ('pickingRadius' in props || 'onLayerClick' in props || 'onLayerHover' in props) {
+      this._setEventHandlingParameters(props);
     }
 
-    if ('width' in parameters || 'height' in parameters) {
-      this.setSize(parameters.width, parameters.height);
+    if ('width' in props || 'height' in props) {
+      this.viewManager.setSize(props.width, props.height);
+      this.width = props.width;
```

Comment:
Does `LayerManager` still need `width` and `height`? Could it introduce discrepancy between layerManager and viewManager?